kura/org.eclipse.kura.driver.opcua.provider/src/main/java/org/eclipse/kura/internal/driver/opcua/OpcUaChannelDescriptor.java

```diff
@@ -33,33 +38,86 @@
  */
 public final class OpcUaChannelDescriptor implements ChannelDescriptor {
 
-    /** Localization Resource. */
-    private static final OpcUaMessages message = LocalizationAdapter.adapt(OpcUaMessages.class);
+    private static final String NODE_ID = "node.id";
+    private static final String NODE_NAMESPACE_INDEX = "node.namespace.index";
+    private static final String NODE_ID_TYPE = "node.id.type";
+
+    private static void addOptions(Tad target, Enum<?>[] values) {
+        final List<Option> options = target.getOption();
+        for (Enum<?> value : values) {
+            Toption option = new Toption();
+            option.setLabel(value.name());
+            option.setValue(value.name());
+            options.add(option);
+        }
+    }
 
     /** {@inheritDoc} */
     @Override
     public Object getDescriptor() {
         final List<Tad> elements = CollectionUtil.newArrayList();
 
         final Tad nodeId = new Tad();
-        nodeId.setName(message.nodeId());
-        nodeId.setId(message.nodeId());
-        nodeId.setDescription(message.nodeId());
+        nodeId.setName(NODE_ID);
+        nodeId.setId(NODE_ID);
+        nodeId.setDescription(NODE_ID);
         nodeId.setType(Tscalar.STRING);
         nodeId.setRequired(true);
         nodeId.setDefault("MyNode");
         elements.add(nodeId);
 
         final Tad namespaceIndex = new Tad();
-        namespaceIndex.setName(message.nodeNamespaceIndex());
-        namespaceIndex.setId(message.nodeNamespaceIndex());
-        namespaceIndex.setDescription(message.nodeNamespaceIndex());
+        namespaceIndex.setName(NODE_NAMESPACE_INDEX);
+        namespaceIndex.setId(NODE_NAMESPACE_INDEX);
+        namespaceIndex.setDescription(NODE_NAMESPACE_INDEX);
         namespaceIndex.setType(Tscalar.INTEGER);
         namespaceIndex.setRequired(true);
         namespaceIndex.setDefault("2");
 
         elements.add(namespaceIndex);
+
+        final Tad nodeIdType = new Tad();
+        nodeIdType.setName(NODE_ID_TYPE);
+        nodeIdType.setId(NODE_ID_TYPE);
+        nodeIdType.setDescription(NODE_ID_TYPE);
+        nodeIdType.setType(Tscalar.STRING);
+        nodeIdType.setRequired(true);
+        nodeIdType.setDefault("STRING");
+
+        addOptions(nodeIdType, NodeIdType.values());
+
+        elements.add(nodeIdType);
         return elements;
     }
 
+    public static int getNodeNamespaceIndex(Map<String, Object> properties) {
+        return Integer.parseInt((String) properties.get(NODE_NAMESPACE_INDEX));
+    }
+
+    public static NodeIdType getNodeIdType(Map<String, Object> properties) {
+        String nodeIdType = (String) properties.get(NODE_ID_TYPE);
+        if (NodeIdType.NUMERIC.name().equals(nodeIdType)) {
```

Comment:
That's not possible, because `NodeIdType.name()` is not considered to be a constant expression by the compiler